src/Field.js

```diff
@@ -1,240 +1,58 @@
-    listener: (state: FieldState) => void
-  ) => {
-    this.unsubscribe = this.props.reactFinalForm.registerField(
-      name,
-      listener,
-      subscription || all,
-      {
-        defaultValue,
-        getValidator: () => this.props.validate,
-        initialValue,
-        isEqual,
-        validateFields
-      }
-    )
-  }
-
-  notify = (state: FieldState) => this.setState({ state })
-
-  componentDidUpdate(prevProps: Props) {
-    const { name, subscription } = this.props
-    if (
-      prevProps.name !== name ||
-      diffSubscription(
-        prevProps.subscription,
-        subscription,
-        fieldSubscriptionItems
-      )
-    ) {
-      if (this.props.reactFinalForm) {
-        // avoid error, warning will alert developer to their mistake
-        this.unsubscribe()
-        this.subscribe(this.props, this.notify)
-      }
-    }
-  }
-
-  componentWillUnmount() {
-    this.unsubscribe()
-  }
-
-  handlers = {
-    onBlur: (event: ?SyntheticFocusEvent<*>) => {
-      const { state } = this.state
-      // this is to appease the Flow gods
-      // istanbul ignore next
-      if (state) {
-        const { format, formatOnBlur } = this.props
-        state.blur()
-        if (format && formatOnBlur) {
-          state.change(format(state.value, state.name))
-        }
-      }
-    },
-    onChange: (event: SyntheticInputEvent<*> | any) => {
-      const { parse, value: _value } = this.props
-
-      // istanbul ignore next
-      if (process.env.NODE_ENV !== 'production' && event && event.target) {
-        const targetType = event.target.type
-        const props: Object = this.props
-        const unknown =
-          ~['checkbox', 'radio', 'select-multiple'].indexOf(targetType) &&
-          !props.type
-
-        const type = targetType === 'select-multiple' ? 'select' : targetType
-        const { value }: any =
-          targetType === 'select-multiple' ? this.state.state || {} : props
-
-        if (unknown) {
-          console.error(
-            `Warning: You must pass \`type="${type}"\` prop to your Field(${
-              props.name
-            }) component.\n` +
-              `Without it we don't know how to unpack your \`value\` prop - ${
-                Array.isArray(value) ? `[${value}]` : `"${value}"`
-              }.`
-          )
-        }
-      }
-
-      const value: any =
-        event && event.target
-          ? getValue(
-              event,
-              this.state.state && this.state.state.value,
-              _value,
-              isReactNative
-            )
-          : event
-      this.state.state &&
-        this.state.state.change(parse ? parse(value, this.props.name) : value)
-    },
-    onFocus: (event: ?SyntheticFocusEvent<*>) => {
-      this.state.state && this.state.state.focus()
-    }
+import useField, { defaultFormat, defaultParse } from './useField'
+
+const Field = ({
+  allowNull,
+  children,
+  component,
+  defaultValue,
+  format = defaultFormat,
```

Comment:
as those are passed directly to `useField` and people might use `useField` directly without ever using `Field` it seems that those defaults should be handled in the `useField` itself

Reply:
I put them there, but then `input.value` was getting formatted to `''` when `undefined`, so I took it back out. Not sure what's best... 🤔